lib/double_entry.rb

```diff
@@ -15,7 +15,9 @@
 require 'double_entry/locking'
 require 'double_entry/transfer'
 require 'double_entry/line'
-require 'double_entry/line_metadata'
+unless DoubleEntry.config.json_metadata
+  require 'double_entry/line_metadata'
+end
```

Comment:
@orien / @swrobel: is there a chicken-and-egg problem here?

The application will `require 'double_entry'` and this check for the `json_metadata` flag will happen before the application sets it. Is that expected?

Reply:
Hmm, good point. Not expected, and I'm not really sure how to handle that appropriately... Any ideas?

Reply:
Great question @ricobl! There's a similar problem here:

https://github.com/envato/double_entry/blob/3faa32ae7501c7c24d9a6ee17f5f9e2252549556/lib/double_entry/line.rb#L59